custom_components/waste_collection_schedule/waste_collection_schedule/source/impactapps_com_au.py

```diff
@@ -0,0 +1,321 @@
+        url = f"{self.api_url}/api/v1/properties.json?street={street_id}"
+        response = session.get(url)
+        response.raise_for_status()
+        properties: List[PropertyResponse] = response.json()["properties"]
+        property_id = next(
+            (item["id"]
+             for item in properties
+             if item["name"] == f"{street_number} {street_name} {suburb}"), None
+        )
+        if property_id is None:
+            raise ValueError(
+                f"Property {street_number} {street_name} {suburb} not found")
+        return property_id
+
+
+class Source:
+    def __init__(self, service: str, property_id: Optional[int] = None, suburb: Optional[str] = None, street_name: Optional[str] = None, street_number: Optional[str] = None):
+
+        if service in SERVICE_MAP_LOOKUP:
+            api_url = SERVICE_MAP_LOOKUP[service]["url"]
+        else:
+            if service.startswith("https://"):
+                api_url = service
+            else:
+                # Assume the service is a council name
+                api_url = f"https://{service}.waste-info.com.au"
+
+        self.api_url = api_url
+        self.property_id = property_id
+
+        if not property_id:
+            if not suburb or not street_name or not street_number:
+                raise ValueError(
+                    "You must provide a property ID or a suburb, street name and street number"
+                )
+            self.suburb = suburb
+            self.street_name = street_name
+            self.street_number = street_number
+            self.location_finder = LocationFinder(self.api_url)
+
+    def fetch(self):
+        start_date = date.today()
+        end_date = start_date + timedelta(365)
+
+        session = requests.Session()
+        session.headers.update(HEADERS)
+
+        if not self.property_id:
+            suburb_id = self.location_finder.find_suburb_id(
+                session, self.suburb)
+            street_id = self.location_finder.find_street_id(
+                session, suburb_id, self.street_name)
+            self.property_id = self.location_finder.find_property_id(
+                session, street_id, self.street_number, self.street_name, self.suburb)
+
+        # Retrieve the collection events for the property
+        url = f"{self.api_url}/api/v1/properties/{self.property_id}.json?start={start_date}&end={end_date}"
+        response = session.get(url)
+        events: List[Union[RecurringEventResponse,
+                           OneOffEventResponse]] = response.json()
+
+        collections: List[Collection] = []
+        for event in events:
+            event_type = event["event_type"]
+            # Skip events that we don't care about (e.g. "special" events)
+            if event_type not in ICON_MAP:
+                continue
+
+            icon = ICON_MAP[event_type]
```

Comment:
I've moved to pulling none when a new waste type is found. It may cause noise with certain "special" waste events (e.g. bulky item and chemical disposals) however these can be easily filtered by the user.